### PR TITLE
Task 129983: Azure Batch task polling doesn't inform REEF when the Sh…

### PR DESCRIPTION
**Pull request by ryumbra**
…im fails to launch

### CODE REVIEW

File: lang/java/reef-runtime-azbatch/src/main/java/org/apache/reef/runtime/azbatch/client/AzureBatchJobSubmissionHandler.java [hunk 1]
```diff
@@ -67,39 +72,46 @@
...
+   * @throws Exception
+   */
   @Override
   public void close() throws Exception {
     LOG.log(Level.INFO, "Closing " + AzureBatchJobSubmissionHandler.class.getName());
   }
 
+  /**
+   * Invoked when JobSubmissionEvent is triggered.
+   * @param jobSubmissionEvent triggered job submission event.
```
Comment (homezcx) [inline lang/java/reef-runtime-azbatch/src/main/java/org/apache/reef/runtime/azbatch/client/AzureBatchJobSubmissionHandler.java]: a blank line is required between comments and tags (param, throws, returns, etc...) as Javadoc convention. But I am not sure if this is enforced by REEF tools.

It applies to line 81, 90, 99 and other files.

http://www.oracle.com/technetwork/java/javase/documentation/index-137868.html#styleguide

File: lang/java/reef-runtime-azbatch/src/main/java/org/apache/reef/runtime/azbatch/driver/AzureBatchEvaluatorShimManager.java [hunk 2]
```diff
@@ -162,102 +186,135 @@ public void onNext(final RemoteMessage<EvaluatorShimProtocol.EvaluatorShimStatus
...
+  /**
+   * Event handler method for {@link ResourceReleaseEvent}. Sends a TERMINATE command to the appropriate evaluator shim.
+   * @param resourceReleaseEvent
+   */
   public void onResourceReleased(final ResourceReleaseEvent resourceReleaseEvent) {
+    String resourceRemoteId = getResourceRemoteId(resourceReleaseEvent.getIdentifier());
+
+    // REEF Common will trigger a ResourceReleaseEvent even if the resource has failed. Since we know that the shim
+    // has already failed, we can safely ignore this.
+    if (this.failedResources.containsKey(resourceReleaseEvent.getIdentifier())) {
```
Comment (homezcx) [inline lang/java/reef-runtime-azbatch/src/main/java/org/apache/reef/runtime/azbatch/driver/AzureBatchEvaluatorShimManager.java]: Is concurrent access issue to failedResources here?

**Review by homezcx — APPROVED** [review]

File: lang/java/reef-runtime-azbatch/src/main/java/org/apache/reef/runtime/azbatch/client/AzureBatchJobSubmissionHandler.java (hunk 1 above)
Comment (ryumbra) [reply lang/java/reef-runtime-azbatch/src/main/java/org/apache/reef/runtime/azbatch/client/AzureBatchJobSubmissionHandler.java]: OK, I'll change those

File: lang/java/reef-runtime-azbatch/src/main/java/org/apache/reef/runtime/azbatch/driver/AzureBatchEvaluatorShimManager.java (hunk 2 above)
Comment (ryumbra) [reply lang/java/reef-runtime-azbatch/src/main/java/org/apache/reef/runtime/azbatch/driver/AzureBatchEvaluatorShimManager.java]: Changed it to remove the element in line 277.

**Pull request merged**